### PR TITLE
Target jack with two and nine

### PR DESCRIPTION
**Pull request by lihaobhsfer**
targeting jacks with two and nine, regular and from seven, player and opponent

some test cases toward the end of test specs might fail due to time out issues, but they pass when run indivually.

### CODE REVIEW

**Comment by lihaobhsfer** [comment]
Just remembered I still need to tweak the styles on jacks with the restructuring. But the functionalities are done :)

File: src/plugins/sails.js [hunk 1]
```diff
@@ -75,6 +75,7 @@ io.socket.on('game', function(evData) {
 				}
 			}
 			break;
+		case 'targetedOneOff':
```
Comment (itsalaidbacklife) [inline src/plugins/sails.js:78]: I don't understand how we were able to play these before without this! Did the countering dialog pop up previously?

File: tests/e2e/specs/game/sevens.spec.js [hunk 2]
```diff
@@ -371,6 +371,66 @@ describe('Playing SEVENS', () => {
...
+			});
+			
+			cy.playOneOffAndResolveAsPlayer(Card.SEVEN_OF_CLUBS);
+	
+			// Play two of spades
+			cy.get('[data-second-card=2-3]')
+				.should('exist')
+				.and('be.visible')
+				.click();
+			// target queen of clubs
```
Comment (itsalaidbacklife) [inline tests/e2e/specs/game/sevens.spec.js:413]: Comment typo: jack of clubs

File: tests/e2e/specs/game/sevens.spec.js [hunk 3]
```diff
@@ -413,6 +473,67 @@ describe('Playing SEVENS', () => {
...
+			});
+
+			cy.playOneOffAndResolveAsPlayer(Card.SEVEN_OF_CLUBS);
+			
+			// Play Nine of diamonds
+			cy.get('[data-top-card=9-1]')
+				.should('exist')
+				.and('be.visible')
+				.click();
+			// target queen of clubs
```
Comment (itsalaidbacklife) [inline tests/e2e/specs/game/sevens.spec.js:516]: Stale comment: should be jack of clubs

File: tests/e2e/support/commands.js [hunk 4]
```diff
@@ -263,6 +263,71 @@ Cypress.Commands.add('playOneOffOpponent', (card) => {
...
+				});
+				break;
+			default:
+				throw new Error(`Error playing ${printCard(card)} as one-off from seven as opponent: invalid target type, ${targetType}`);
+			}
+			if (!foundCard) {
+				throw new Error(`Error playing targeted one-off as opponent: could not find ${printCard(card)} in opponent hand`);
+			}
+			if (!foundTarget) {
+				throw new Error(`Error playing targeted one-off as opponent: could not find ${printCard(target)} in player hand`);
```
Comment (itsalaidbacklife) [inline tests/e2e/support/commands.js:312]: Cannot find target in player field*

**Review by itsalaidbacklife — CHANGES_REQUESTED** [review]
Looks good; requesting small chores: logs, comments, error msg

File: src/plugins/sails.js (hunk 1 above)
Comment (lihaobhsfer) [reply src/plugins/sails.js:78]: I noticed this when writing a test case from opponent's perspective. previously with targeted one-offs the test cases were only from player's perspective and the game probably didn't need to check if there was a coutering dialog or not.

**Review by itsalaidbacklife — APPROVED** [review]
Nice work. Lasts moves are done!!!!!

**Pull request merged**